spring-cloud-app-broker-acceptance-tests/src/test/java/org.springframework.cloud.appbroker.acceptance/UpdateInstanceWithServicesAcceptanceTest.java

```diff
@@ -0,0 +1,89 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.appbroker.acceptance;
+
+import org.cloudfoundry.operations.applications.ApplicationSummary;
+import org.cloudfoundry.operations.services.ServiceInstanceSummary;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class UpdateInstanceWithServicesAcceptanceTest extends CloudFoundryAcceptanceTest {
+
+	private static final String APP_NAME = "app-update-services";
+	private static final String SI_NAME = "si-update-services";
+
+	private static final String BACKING_SERVICE_NAME = "backing-service-update";
+	private static final String BACKING_SI_NAME = "backing-service-instance-update";
+
+	@BeforeEach
+	void setUpServiceBrokerForService() {
+		deployServiceBrokerForService(BACKING_SERVICE_NAME);
+	}
+
+	@AfterEach
+	void tearDownServiceBrokerForService() {
+		deleteServiceBrokerForService(BACKING_SERVICE_NAME);
+	}
+
+	@Test
+	@AppBrokerTestProperties({
+		"spring.cloud.appbroker.services[0].service-name=example",
+		"spring.cloud.appbroker.services[0].plan-name=standard",
+		"spring.cloud.appbroker.services[0].apps[0].name=" + APP_NAME,
+		"spring.cloud.appbroker.services[0].apps[0].path=classpath:demo.jar",
+
+		"spring.cloud.appbroker.services[0].apps[0].services[0].service-instance-name=" + BACKING_SI_NAME,
+
+		"spring.cloud.appbroker.services[0].services[0].service-instance-name=" + BACKING_SI_NAME,
+		"spring.cloud.appbroker.services[0].services[0].name=" + BACKING_SERVICE_NAME,
+		"spring.cloud.appbroker.services[0].services[0].plan=standard",
+	})
+	void shouldPushAppWithServicesBind() {
+		// when a service instance is created
+		createServiceInstance(SI_NAME);
+
+		Optional<ServiceInstanceSummary> serviceInstance = getServiceInstance(SI_NAME);
+		assertThat(serviceInstance).hasValueSatisfying(value ->
+			assertThat(value.getLastOperation()).contains("completed"));
+
+		// then a backing application is deployed
+		Optional<ApplicationSummary> backingApplication = getApplicationSummaryByName(APP_NAME);
+		assertThat(backingApplication).hasValueSatisfying(app ->
+			assertThat(app.getRunningInstances()).isEqualTo(1));
+
+		Map<String, Object> parameters = new HashMap<>();
+		parameters.put("parameter1", "value1");
+		parameters.put("parameter2", "value2");
+		parameters.put("parameter3", "value3");
+		updateServiceInstance(SI_NAME, parameters);
+
+		// the services are bound to it
+		Optional<ServiceInstanceSummary> serviceInstance1 = getServiceInstance(BACKING_SI_NAME);
+		assertThat(serviceInstance1).hasValueSatisfying(instance ->
+			assertThat(instance.getApplications()).contains(APP_NAME));
+
```

Comment:
Shouldn't we assert that the service was updated?

Reply:
I couldn't figure out how to do that, since there's no before/after properties to query on the backing service. I'll think about it more. 

I'm planning to remove pushing a second app broker and using that for a backing SI broker, as pushing that that second broker app and then having it push additional backing apps takes a lot of time. So I don't want to rely on that mechanism for checking before/after state of a backing SI.